runtime/j9vm/j9scar.tdf

```diff
@@ -348,3 +348,5 @@ TraceEntry=Trc_SC_UnloadLibrary_Entry NoEnv Overhead=1 Level=2 Template="JVM_Unl
 TraceEvent=Trc_SC_LoadLibrary_BootStrap NoEnv Overhead=1 Level=3 Template="JVM_LoadLibrary(name=%s) Bootstrap library"
 TraceEvent=Trc_SC_LoadLibrary_OpenShared NoEnv Overhead=1 Level=3 Template="JVM_LoadLibrary(name=%s) j9sl_open_shared_library"
 TraceEvent=Trc_SC_LoadLibrary_OpenShared_Decorate NoEnv Overhead=1 Level=3 Template="JVM_LoadLibrary(name=%s) j9sl_open_shared_library J9PORT_SLOPEN_DECORATE"
+
+TraceExit=Trc_SC_LoadLibrary_Exit2 NoEnv Overhead=1 Level=2 Template="JVM_LoadLibrary -- return 0x%zx"
```

Comment:
This is a fix to the original tracepoint, the parameters didn't change, it can be done in place rather than creating a new tracepoint. Unless there is some older version of the code that passes an `int` `result` to the tracepoint.

Reply:
`Trc_SC_LoadLibrary_Exit` took parameters `HINSTANCE dllHandle` before.

As per https://github.com/eclipse-openj9/openj9/blob/master/doc/diagnostics/AddingTracepoints.md,
_If a tracepoint's signature changes, obsolete it and create a new one. By signature we mean the types, order and total number of format specifiers in the Template. Changes to the Tracepoint Type, Overhead, Level and NoEnv parameters, and cosmetic changes to the text in the Template can be made without adding a new tracepoint._

This PR changes the format type specifier, creating a new one is less risky than reusing original tracepoint.

Reply:
The signature didn't change, it was incorrect which is a bug, and it can be fixed in place. i.e. if you use the new tracepoint to format older data, it will work properly, which is the real criteria.

Reply:
> creating a new one is less risky than reusing original tracepoint

It fixes formatting for newer jvms, but doesn't fix the bug when formatting tracepoints for older jvms.

Reply:
The signature does change.
Built three VM, first one is existing code w/o this PR, second one is to modify existing `Trc_SC_LoadLibrary_Exit`, third is w/ this PR.
The snaptrace file generated by first VM can be parse correctly by third VM, and produce same result as same VM (first one).
```
0x0000000011bfd400 j9scar.115          Exit       <JVM_LoadLibrary -- return -276760736
```
When second VM parses the sanptrace file, it produces 1 warnings, and the trace message looks like
```
0x0000000011bfd400 j9scar.115          Exit       <JVM_LoadLibrary -- return 0x<missing in trace>
```

Reply:
Maybe it doesn't work how I thought it does. Can I look at the commit for the second VM build?

What do you get parsing a snaptrace from the 3rd VM, using the 3rd VM to parse?

Reply:
Also what do you get parsing a snaptrace from the 2nd VM, using the 2nd VM to parse?

Reply:
It is a local build, so no commit link.

> What do you get parsing a snaptrace from the 3rd VM, using the 3rd VM to parse?

```
0x0000000017fae400 j9scar.232          Exit       <JVM_LoadLibrary -- return 0x7fd33dc2ddb0
```

> Also what do you get parsing a snaptrace from the 2nd VM, using the 2nd VM to parse?

```
0x000000001680e400 j9scar.115          Exit       <JVM_LoadLibrary tptwo -- return 0x7fa9dae2af20
```

Note:
_Setting the Obsolete keyword guards against usage of the old tracepoint in the current code stream, but keeps it available for the formatter to use when processing trace from older releases._

Reply:
ok

Reply:
@keithc-ca can you look at this pls, it seems weird to me.

Reply:
The name of the tracepoint does not need to change if the old one is marked as `Obsolete`.
Otherwise I don't understand what might be 'weird'.

Reply:
https://github.com/eclipse-openj9/openj9/pull/12981#discussion_r707462904

Reply:
Why is a new tracepoint required, why can't the incorrect signature in the original tracepoint be fixed in place?

Reply:
Changing the signature of an existing tracepoint would incorrectly interpret existing tracefiles (which captured an `int` instead of a pointer).